<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Assignment12Class</class>
 <widget class="QMainWindow" name="Assignment12Class">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>551</width>
    <height>565</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Assignment12</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout_3">
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>1</number>
      </property>
      <widget class="QWidget" name="tab_17">
       <attribute name="title">
        <string>mode A</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QListWidget" name="dogListWidget">
          <property name="font">
           <font>
            <pointsize>12</pointsize>
           </font>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QFormLayout" name="formLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="label">
            <property name="font">
             <font>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Name</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="nameLineEdit"/>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_3">
            <property name="font">
             <font>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Breed</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="breedLineEdit"/>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_2">
            <property name="font">
             <font>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Date Of Birth</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QLineEdit" name="dateLineEdit"/>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_4">
            <property name="font">
             <font>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Vaccination Count</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QLineEdit" name="vaccinationLineEdit"/>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="label_5">
            <property name="font">
             <font>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Photograph</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QLineEdit" name="photographLineEdit"/>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="0" column="0">
           <widget class="QPushButton" name="addButton">
            <property name="font">
             <font>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Add</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1" colspan="2">
           <widget class="QPushButton" name="deleteButton">
            <property name="font">
             <font>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Delete</string>
            </property>
           </widget>
          </item>
          <item row="0" column="3">
           <widget class="QPushButton" name="updateButton">
            <property name="font">
             <font>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Update</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0" colspan="2">
           <widget class="QPushButton" name="undoButton">
            <property name="font">
             <font>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Undo</string>
            </property>
            <property name="shortcut">
             <string>Ctrl+Z</string>
            </property>
           </widget>
          </item>
          <item row="1" column="2" colspan="2">
           <widget class="QPushButton" name="redoButton">
            <property name="font">
             <font>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Redo</string>
            </property>
            <property name="shortcut">
             <string>Ctrl+Y</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_18">
       <attribute name="title">
        <string>mode B</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QListWidget" name="dogListWidget_modeB">
          <property name="font">
           <font>
            <pointsize>12</pointsize>
           </font>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QGridLayout" name="gridLayout">
          <item row="0" column="0">
           <widget class="QPushButton" name="nextButton">
            <property name="font">
             <font>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Next</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QPushButton" name="saveButton">
            <property name="font">
             <font>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Save</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QFormLayout" name="formLayout_3">
          <item row="0" column="0">
           <widget class="QLabel" name="label_6">
            <property name="font">
             <font>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Breed</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="breedLineEdit_modeB">
            <property name="font">
             <font>
              <pointsize>8</pointsize>
             </font>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_7">
            <property name="font">
             <font>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Vaccination Count</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="vaccinationLineEdit_modeB">
            <property name="font">
             <font>
              <pointsize>8</pointsize>
             </font>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QPushButton" name="filterButton">
          <property name="font">
           <font>
            <pointsize>12</pointsize>
           </font>
          </property>
          <property name="contextMenuPolicy">
           <enum>Qt::NoContextMenu</enum>
          </property>
          <property name="text">
           <string>Filter</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QListWidget" name="mylistWidget">
          <property name="font">
           <font>
            <pointsize>12</pointsize>
           </font>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QPushButton" name="undoBButton">
            <property name="font">
             <font>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Undo</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="redoBButton">
            <property name="font">
             <font>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Redo</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QPushButton" name="newWindowButton">
          <property name="font">
           <font>
            <pointsize>12</pointsize>
           </font>
          </property>
          <property name="text">
           <string>Display mylist in new window</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="displayButton">
          <property name="font">
           <font>
            <pointsize>12</pointsize>
           </font>
          </property>
          <property name="text">
           <string>Display mylist with external app</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>551</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="Assignment12.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>addButton</sender>
   <signal>clicked()</signal>
   <receiver>Assignment12Class</receiver>
   <slot>addDog()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>121</x>
     <y>488</y>
    </hint>
    <hint type="destinationlabel">
     <x>374</x>
     <y>461</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>updateButton</sender>
   <signal>clicked()</signal>
   <receiver>Assignment12Class</receiver>
   <slot>updateDog()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>527</x>
     <y>488</y>
    </hint>
    <hint type="destinationlabel">
     <x>374</x>
     <y>506</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>saveButton</sender>
   <signal>clicked()</signal>
   <receiver>Assignment12Class</receiver>
   <slot>saveDog()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>511</x>
     <y>216</y>
    </hint>
    <hint type="destinationlabel">
     <x>374</x>
     <y>234</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>displayButton</sender>
   <signal>clicked()</signal>
   <receiver>Assignment12Class</receiver>
   <slot>runMyList()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>219</x>
     <y>522</y>
    </hint>
    <hint type="destinationlabel">
     <x>374</x>
     <y>495</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>filterButton</sender>
   <signal>clicked()</signal>
   <receiver>Assignment12Class</receiver>
   <slot>filter()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>267</x>
     <y>304</y>
    </hint>
    <hint type="destinationlabel">
     <x>374</x>
     <y>342</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>nextButton</sender>
   <signal>clicked()</signal>
   <receiver>Assignment12Class</receiver>
   <slot>next()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>103</x>
     <y>216</y>
    </hint>
    <hint type="destinationlabel">
     <x>8</x>
     <y>206</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>undoButton</sender>
   <signal>clicked()</signal>
   <receiver>Assignment12Class</receiver>
   <slot>undo()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>130</x>
     <y>521</y>
    </hint>
    <hint type="destinationlabel">
     <x>363</x>
     <y>514</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>redoButton</sender>
   <signal>clicked()</signal>
   <receiver>Assignment12Class</receiver>
   <slot>redo()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>527</x>
     <y>521</y>
    </hint>
    <hint type="destinationlabel">
     <x>349</x>
     <y>519</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>newWindowButton</sender>
   <signal>clicked()</signal>
   <receiver>Assignment12Class</receiver>
   <slot>showNewWindow()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>457</x>
     <y>467</y>
    </hint>
    <hint type="destinationlabel">
     <x>513</x>
     <y>554</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>undoBButton</sender>
   <signal>clicked()</signal>
   <receiver>Assignment12Class</receiver>
   <slot>undo_modeB()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>72</x>
     <y>431</y>
    </hint>
    <hint type="destinationlabel">
     <x>545</x>
     <y>401</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>redoBButton</sender>
   <signal>clicked()</signal>
   <receiver>Assignment12Class</receiver>
   <slot>redo_modeB()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>421</x>
     <y>444</y>
    </hint>
    <hint type="destinationlabel">
     <x>546</x>
     <y>446</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>addDog()</slot>
  <slot>updateDog()</slot>
  <slot>saveDog()</slot>
  <slot>runMyList()</slot>
  <slot>filter()</slot>
  <slot>next(vector)</slot>
  <slot>next()</slot>
  <slot>undo()</slot>
  <slot>redo()</slot>
  <slot>showNewWindow()</slot>
  <slot>undo_modeB()</slot>
  <slot>redo_modeB()</slot>
 </slots>
</ui>
